### PR TITLE
[HotFix] Fix Colorful mock for `with_style`

### LINKED ISSUE #14849: Ray Start Fails if colorful is not installed
### What is the problem?
When I run  `ray start --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml)'` without `colorful` installed, I get the following error:
```
Local node IP: 10.138.0.22
2021-03-22 16:56:36,806 INFO services.py:1269 -- View the Ray dashboard at http://127.0.0.1:8265

--------------------
Ray runtime started.
--------------------

Next steps
  To connect to this Ray runtime from another node, run
    ray start --address='10.138.0.22:6379' --redis-password='5241590000000000'
  
  Alternatively, use the following Python code:
Traceback (most recent call last):
  File "/opt/conda/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/ray/scripts/scripts.py", line 1706, in main
    return cli()
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/ray/scripts/scripts.py", line 586, in start
    with cf.with_style("monokai") as c:
AttributeError: __enter__
```

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* The mock for `with_style` should yield a contextmanager, not a function.

## Related issue number

Closes #14849

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/cli_logger.py [hunk 1]
```diff
@@ -51,6 +51,14 @@ def disable(self):
...
 
         def __getattr__(self, name):
+            if name == "with_style":
+                @contextmanager
+                def wrapper(x):
+                    class identityClass:
+                        def __getattr__(self, name):
+                            return lambda y: y
+                    yield identityClass()
+                return wrapper
```
Comment (richardliaw) [inline python/ray/autoscaler/_private/cli_logger.py:61]: Can you put this into a separate function, like:

```

@contextmanager
def with_style(self, style_name):
    class IdentityClass:
        ...
    yield IdentityClass()

def __getattr__(self, name):
    ...

Comment (ijrsvt) [reply python/ray/autoscaler/_private/cli_logger.py]: Sure!

**Comment by ijrsvt** [comment]
cc @richardliaw I added testing :)

File: python/ray/tests/test_cli_logger.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from ray.autoscaler._private import cli_logger
```
Comment (richardliaw) [inline python/ray/tests/test_cli_logger.py:1]: you'll need to add this to BUILD, but we can do that in a separate PR

**Review by richardliaw — APPROVED** [review]

File: python/ray/tests/test_cli_logger.py (hunk 2 above)
Comment (ijrsvt) [reply python/ray/tests/test_cli_logger.py:1]: Added here, because of Lint failure :'(
